scripts/build-dist.sh

```diff
@@ -16,25 +17,33 @@ case "$(uname -s)" in
     ;;
 esac
 
+version=`node -e "var fs=require('fs');console.log(JSON.parse(fs.readFileSync('package.json')).version)"`
+
 rm -rf artifacts dist
-rm -rf dist
 mkdir artifacts
-mkdir dist{,/bin,/lib}
+mkdir dist{,/bin,/lib,/lib-legacy}
 
 # Workaround for https://github.com/yarnpkg/yarn/issues/2591
 eval $system_yarn run build
 eval $system_yarn run build-bundle
 chmod +x artifacts/*.js
+# Verify that it works as expected
+[[ "$version" == "$(node artifacts/yarn-$version.js --version)" ]] || exit 1
+[[ "$version" == "$(node artifacts/yarn-legacy-$version.js --version)" ]] || exit 1
 
 cp package.json dist/
 cp README.md dist/
 cp LICENSE dist/
-cp artifacts/yarn-legacy-*.js dist/lib/yarn-cli.js
-cp bin/yarn-bundle-entry.js dist/bin/yarn.js
-cp bin/{yarn,yarnpkg,*.cmd} dist/bin/
+cp artifacts/yarn-legacy-$version.js dist/lib-legacy/cli.js
+cp artifacts/yarn-$version.js dist/lib/cli.js
```

Comment:
Hmm, including both the legacy **and** the non-legacy version is going to double the on-disk footprint for Yarn. The Node Docker people aren't going to be happy about that as it'll add an extra 3 MB to their image, which is significant given the Alpine image is only ~20 MB or so 😃

Can we **just** bundle the legacy one, like before? I don't really see any major advantages to building and bundling both.